src/DataProviders/SqlDataProvider/SqlDataProvider.ts

```diff
@@ -1,61 +1,83 @@
-﻿//import sql from "mssql";
-const sql = require('mssql');
-import * as assert from "assert";
-import { IDataProvider } from "../IDataProvider";
-import { ConnectionPool, ConnectionError, config, IResult } from "mssql";
-import { SeasonInfo } from "../../DataModels/SeasonInfo";
+﻿import * as assert from "assert";
+import { inject, injectable } from "inversify";
+import { ConnectionPool } from "mssql";
+import { isNullOrUndefined } from "util";
+import { Configuration } from "../../Common/Configuration";
+import { SeasonInfo } from "../../Models/SeasonInfo";
+import { TYPES } from "../../types";
+import { IDataProvider, IDataProviderOptions } from "../IDataProvider";
+import { HazzatDbSchema } from "./HazzatDbSchema";
+
+/**
+ * Sql Data Provider Options
+ */
+@injectable()
+export class SqlDataProviderOptions implements IDataProviderOptions {
+    // TODO: Is there a way to pass this in from App?
```

Comment:
what is the concern here?

Reply:
btw, if you want to reduce the number of classes to maintain, you can make `Configuration` itself injectable, and directly use `dbConnectionString` from there.

Reply:
I didn't want all my classes to take a dependency on configuration.  I wanted only the app.ts to have that dependency, and it would initialize the subsequent classes accordingly.

Reply:
That sounds like a good idea initially, but if you think about it, they all depend on the configuration anyway - it's just that your app now stands between them and needs to do a bunch of translation. I don't think it's that bad to depend on configuration in this case (and I've seen that pattern frequently), but it's up to you :)